src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -106,7 +106,10 @@ Copyright (c) .NET Foundation. All rights reserved.
                  ResourceName="CannotHaveSingleFileWithoutRuntimeIdentifier" />
     <NETSdkError Condition="'$(PublishSingleFile)' == 'true' and '$(UseAppHost)' != 'true'"
                  ResourceName="CannotHaveSingleFileWithoutAppHost" />
-
+    <NETSdkError Condition="'$(UseWindowsForms)' == 'true' and '$(PublishTrimmed)' == 'true' and '$(BuiltInComSupport)' != 'true'"
```

Comment:
What does it mean when `BuiltInComSupport` is the empty string (which is the default as written)? We should be consistent whether trimming or not.

I think it would be better to check `'$(BuiltInComSupport)' == 'false'`, and set it to false when trimming like we do for startup hooks:
https://github.com/dotnet/sdk/blob/9da281a438aa7cf9b80cd313f66917b87a4bc72e/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets#L27-L31

(The alternative would be to keep the `!= 'true'` and set `BuiltInComSupport` to `true` when not trimming.)

Reply:
This is a little tricky, since for BUILD we want all WinForm apps to get this warning. There is a parallel effort to add feature switch support for BuiltInComSupport but that might not get there by BUILD. As is, setting this property to 'true' for WinForm Apps will allow to go to trimming even though it is a no-op if the build-in COM support doesn't get in by BUILD.